cpp/src/sampling/detail/sampling_utils_impl.cuh

```diff
@@ -285,5 +302,283 @@ sample_edges(raft::handle_t const& handle,
+          handle,
+          graph_view,
+          vertex_frontier.bucket(0),
+          edge_src_dummy_property_t{}.view(),
+          edge_dst_dummy_property_t{}.view(),
+          edge_dummy_property_t{}.view(),
+          sample_edges_op_t<vertex_t>{},
+          rng_state,
+          fanout,
+          with_replacement,
+          std::make_optional<thrust::tuple<vertex_t, vertex_t>>(
+            cugraph::invalid_vertex_id<vertex_t>::value,
+            cugraph::invalid_vertex_id<vertex_t>::value),
+          true);
+      std::tie(majors, minors) = std::move(sample_e_op_results);
+    }
+  }
+
+  if (weights) {
+    auto edge_first = thrust::make_zip_iterator(
+      thrust::make_tuple(majors.begin(), minors.begin(), (*weights).begin()));
+    auto end_iter =
+      thrust::remove_if(handle.get_thrust_policy(),
+                        edge_first,
+                        edge_first + majors.size(),
+                        [] __device__(auto tuple) {
+                          auto v1 = thrust::get<0>(tuple);
+                          auto v2 = thrust::get<1>(tuple);
+
+                          return ((v1 == cugraph::invalid_vertex_id<vertex_t>::value) ||
+                                  (v1 == cugraph::invalid_vertex_id<vertex_t>::value));
+                        });
+
+    size_t new_size = thrust::distance(edge_first, end_iter);
+
+    if (new_size != majors.size()) {
+      majors.resize(new_size, handle.get_stream());
+      majors.shrink_to_fit(handle.get_stream());
+      minors.resize(new_size, handle.get_stream());
+      minors.shrink_to_fit(handle.get_stream());
+      (*weights).resize(new_size, handle.get_stream());
+      (*weights).shrink_to_fit(handle.get_stream());
+    }
+  } else {
+    auto edge_first = thrust::make_zip_iterator(thrust::make_tuple(majors.begin(), minors.begin()));
+    auto end_iter =
+      thrust::remove_if(handle.get_thrust_policy(),
+                        edge_first,
+                        edge_first + majors.size(),
+                        [] __device__(auto tuple) {
+                          auto v1 = thrust::get<0>(tuple);
+                          auto v2 = thrust::get<1>(tuple);
+
+                          return ((v1 == cugraph::invalid_vertex_id<vertex_t>::value) ||
+                                  (v1 == cugraph::invalid_vertex_id<vertex_t>::value));
+                        });
+
+    size_t new_size = thrust::distance(edge_first, end_iter);
+
+    if (new_size != majors.size()) {
+      majors.resize(new_size, handle.get_stream());
+      majors.shrink_to_fit(handle.get_stream());
+      minors.resize(new_size, handle.get_stream());
+      minors.shrink_to_fit(handle.get_stream());
+    }
+  }
+
+  return std::make_tuple(std::move(majors),
+                         std::move(minors),
+                         std::move(edge_ids),
+                         std::move(weights),
+                         std::move(edge_types),
+                         std::move(labels));
```

Comment:
It seems like `labels` is not property updated.

Reply:
Fixed in latest push